### PR TITLE
Remove the Managed VM feature flag and checks

### LINKED ISSUE #1032: remove feature flag checks from fully GA'd features
app engine deployment and cloud debugging are now turned on for all users. We should remove the feature flag checks and move whatever we can into the plugin.xml

### PR DESCRIPTION
**Pull request by patflynn**
Closes #1032

### CODE REVIEW

**Comment by patflynn** [comment]
@aslo @etanshaul

**Comment by joaoandremartins** [comment]
Should we also set to false, or remove the flag, where we were setting it before? Or was it set to true by default?

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/GctFeature.java [hunk 1]
```diff
@@ -24,16 +24,34 @@
 
 /**
  * The set of Google Cloud Tools features that we want to be able to conditionally enable.
+ *
+ * <p>Enum elements marked {@link Deprecated} are fully released and no longer hidden behind a flag.
  */
 public enum GctFeature implements Feature {
+  @Deprecated
   APPENGINE_FLEX(null, "feature.appengine.flex", "ct4ij.feature.appengine.flex");
```
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/GctFeature.java]: Why can't we just remove this line? What could be broken?

Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/GctFeature.java]: @joaoandremartins nothing. The main reason I left it is that it serves as an example, provides history, and the enum looks weird with no instances.  I'm on the fence about keeping the old ones around.

Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/GctFeature.java]: Ack. Could be good to add a clarifying "// Left here just to serve as an example. Don't use." comment. I think that's why you added the @Deprecated annotation for.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/startup/CloudToolsPluginInitializationComponent.java [hunk 2]
```diff
@@ -61,38 +47,13 @@ public void initComponent() {
...
 
     new ConflictingAppEnginePluginCheck().notifyIfConflicting();
   }
 
-  private void initAppEngineSupport(
-      CloudToolsPluginConfigurationService pluginConfigurationService) {
-    AppEngineCloudType appEngineCloudType = new AppEngineCloudType();
-    pluginConfigurationService.registerExtension(ServerType.EP_NAME, appEngineCloudType);
-    pluginConfigurationService.registerExtension(ConfigurationType.CONFIGURATION_TYPE_EP,
-        new DeployToServerConfigurationType(appEngineCloudType));
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/startup/CloudToolsPluginInitializationComponent.java]: Is this extension point also handled by the additions to plugin.xml? I see the line item for `AppEngineCloudType` there, but not `DeployToServerConfigurationType`.

**Comment by patflynn** [comment]
Yeah, I remember having to do a fair bit of spelunking back in the day to
figure out I had to register this manually when I don't use the
plugin.xml.  Still testing to make sure I haven't broken anything. but so
far looks good.

**Pull request merged**